### PR TITLE
Chapter 4 and 5

### PR DESCRIPTION
**Pull request by nuriman1**

### CODE REVIEW

**Comment by nuriman1** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  line-height: 1.7em;
+}
+
+/* header */
+
+#logo {
+ float: left;
+ margin-right: 10px;
+ font-size: 1.7em;
+ color: #fff;
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]: check the color code, please

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+ text-transform: uppercase;
+ letter-spacing: -1px;
+ padding-top: 9px;
+ font-weight: bold;
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
 font-weight: bold;
 
}
```

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  }
+  ul {
+    float: right;
+    list-style: none;
+    li {
+      float: left;
+      margin-left: 15px;
+    }
+  }
+}
```
Comment (saobv-2223) [inline app/assets/stylesheets/custom.scss]:
```suggestion
footer {
  margin-top: 45px;
  padding-top: 5px;
  border-top: 1px solid #eaeaea;
  color: #777;

  a {
    color: #555;

    &:hover {
      color: #222;
    }
  }

  small {
    float: left;
  }

  ul {
    float: right;
    list-style: none;

    li {
      float: left;
      margin-left: 15px;
    }
  }
}
```

File: app/assets/stylesheets/users.scss [hunk 4]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
+// They will automatically be included in application.css.
+// You can use Sass (SCSS) here: https://sass-lang.com/
```
Comment (saobv-2223) [inline app/assets/stylesheets/users.scss]: remove file that never used

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,4 @@
+class UsersController < ApplicationController
```
Comment (saobv-2223) [inline app/controllers/users_controller.rb]: in this chapter, we dont use UsersController, therefore, delete it

File: app/views/layouts/_footer.html.erb [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+		by <%= link_to "Michael Hartl", "https://www.michaelhartl.com/" %>
+	</small>
+	<nav>
+		<ul>
+			<li><%= link_to "About", about_path %></li>
+			<li><%= link_to "Contact", contact_path %></li>
+			<li><%= link_to "News", "https://news.railstutorial.org/" %>
+		</ul>
+	</nav>
+</footer>
```
Comment (saobv-2223) [inline app/views/layouts/_footer.html.erb]:
```suggestion
<footer class="footer">
  <small>
    The <%= link_to "Ruby on Rails Tutorial", "https://www.railstutorial.org/" %>
    by <%= link_to "Michael Hartl", "https://www.michaelhartl.com/" %>
  </small>
  <nav>
    <ul>
      <li><%= link_to "About", about_path %></li>
      <li><%= link_to "Contact", contact_path %></li>
      <li><%= link_to "News", "https://news.railstutorial.org/" %>
    </ul>
  </nav>
</footer>
```

File: app/views/static_pages/help.html.erb [hunk 7]
```diff
@@ -3,8 +3,8 @@
 <h1>Help</h1>
 <p>
   Get help on the Ruby on Rails Tutorial at the 
-  <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
+  <%= link_to "Rails Tutorial help page", "http://www.railstutorial.org/help" %>
```
Comment (saobv-2223) [inline app/views/static_pages/help.html.erb]: please, call link from settings.yml file. 
```suggestion
  <%= link_to "Rails Tutorial help page", Settings.links.rails_tutorial.help_page %>
```

File: config/routes.rb [hunk 8]
```diff
@@ -1,7 +1,9 @@
 Rails.application.routes.draw do
+  get 'users/new'
```
Comment (saobv-2223) [inline config/routes.rb]: using " please

File: app/assets/stylesheets/custom.scss (hunk 2 above)
Comment (nuriman1) [reply app/assets/stylesheets/custom.scss]: done

File: app/assets/stylesheets/custom.scss (hunk 3 above)
Comment (nuriman1) [reply app/assets/stylesheets/custom.scss]: done

File: app/assets/stylesheets/users.scss (hunk 4 above)
Comment (nuriman1) [reply app/assets/stylesheets/users.scss]: done

File: app/controllers/users_controller.rb (hunk 5 above)
Comment (nuriman1) [reply app/controllers/users_controller.rb]: done

File: app/views/layouts/_footer.html.erb (hunk 6 above)
Comment (nuriman1) [reply app/views/layouts/_footer.html.erb]: done

File: config/routes.rb (hunk 8 above)
Comment (nuriman1) [reply config/routes.rb]: done

File: app/views/static_pages/help.html.erb (hunk 7 above)
Comment (nuriman1) [reply app/views/static_pages/help.html.erb]: done

File: config/routes.rb (hunk 8 above)
Comment (nuriman1) [reply config/routes.rb]: done

File: app/assets/stylesheets/custom.scss (hunk 1 above)
Comment (nuriman1) [reply app/assets/stylesheets/custom.scss]: done

**Comment by nuriman1** [comment]
ready

File: app/assets/stylesheets/custom.scss [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+  text-align: center;
+}
+
+h2 {
+  font-size: 1.2em;
+  letter-spacing: -1px;
+  margin-bottom: 30px;
+  text-align: center;
+  font-weight: normal;
+  color: #777;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: add color variables in other file, then call back here

File: app/assets/stylesheets/custom.scss [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+  line-height: 1.7em;
+}
+
+/* header */
+
+#logo {
+ float: left;
+ margin-right: 10px;
+ font-size: 1.7em;
+ color: #0006ff;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: similar

File: app/assets/stylesheets/custom.scss [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+ text-transform: uppercase;
+ letter-spacing: -1px;
+ padding-top: 9px;
+ font-weight: bold;
+
+}
+#logo:hover {
+  color: #F90E05;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: similar

File: app/assets/stylesheets/custom.scss [hunk 12]
```diff
@@ -0,0 +1,97 @@
...
+  text-decoration: none;
+}
+
+/*footer*/
+
+footer {
+  margin-top: 45px;
+  padding-top: 5px;
+  border-top: 1px solid #eaeaea;
+  color: #777;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: similar

File: app/assets/stylesheets/custom.scss [hunk 13]
```diff
@@ -0,0 +1,97 @@
...
+
+  a {
+    color: #555;
+
+    &:hover{
+      color: #222;
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]: similar

**Review by chungpv-1008 — COMMENTED** [review]
You fix comment, then merger the pull request yourself

File: app/assets/stylesheets/custom.scss (hunk 9 above)
Comment (nuriman1) [reply app/assets/stylesheets/custom.scss]: ok done

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**